### PR TITLE
docs(NODE-4724): update fle docs to use "in use encryption" terminology

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

Docs

#### What is the motivation for this change?

Consistency across documentation

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

**Review by durran — CHANGES_REQUESTED** [review]
1 more: https://github.com/mongodb/node-mongodb-native/blob/8f8b480a6aeed89ea40f034b16c6b26355959781/src/connection_string.ts#L1262

**Review by durran — APPROVED** [review]

File: src/collection.ts [hunk 1]
```diff
@@ -288,7 +288,7 @@ export class Collection<TSchema extends Document = Document> {
       options = {};
     }
 
-    // CSFLE passes in { w: 'majority' } to ensure the lib works in both 3.x and 4.x
+    // In use encryption passes in { w: 'majority' } to ensure the lib works in both 3.x and 4.x
```
Comment (dariakp) [inline src/collection.ts]: I think this should refer to the actual module name, instead of an encryption type for clarity

File: src/deps.ts [hunk 2]
```diff
@@ -288,7 +288,7 @@ export interface AutoEncryptionOptions {
    *
    * **NOTE**: Supplying options.schemaMap provides more security than relying on JSON Schemas obtained from the server.
    * It protects against a malicious server advertising a false JSON Schema, which could trick the client into sending decrypted data that should be encrypted.
-   * Schemas supplied in the schemaMap only apply to configuring automatic encryption for client side encryption.
+   * Schemas supplied in the schemaMap only apply to configuring automatic encryption for in use encryption.
```
Comment (dariakp) [inline src/deps.ts:291]: is `schemaMap` applicable to QE? if not, "CSFLE" or "Client-Side Field Level Encryption" would be more appropriate here

File: src/error.ts [hunk 3]
```diff
@@ -111,7 +111,7 @@ export interface ErrorDescription extends Document {
  * @category Error
  *
  * @privateRemarks
- * CSFLE has a dependency on this error, it uses the constructor with a string argument
+ * In use encryption has a dependency on this error, it uses the constructor with a string argument
```
Comment (dariakp) [inline src/error.ts]: This should probably refer to the module as well, unless we mean that something about QE/CSFLE specifically has a dependency on this

File: src/error.ts [hunk 4]
```diff
@@ -568,7 +568,7 @@ export class MongoNetworkError extends MongoError {
  * @category Error
  *
  * @privateRemarks
- * CSFLE has a dependency on this error with an instanceof check
+ * In use encryption has a dependency on this error with an instanceof check
```
Comment (dariakp) [inline src/error.ts]: same comment as the other in this file

File: src/collection.ts [hunk 5]
```diff
@@ -288,8 +288,8 @@ export class Collection<TSchema extends Document = Document> {
       options = {};
     }
 
-    // CSFLE passes in { w: 'majority' } to ensure the lib works in both 3.x and 4.x
-    // we support that option style here only
+    // Older versions of mongodb-client-encryption passes in hardcoded { w: 'majority' }
```
Comment (dariakp) [inline src/collection.ts:291]:
```suggestion
    // Older versions of mongodb-client-encryption pass in hardcoded { w: 'majority' }
```

File: src/collection.ts [hunk 6]
```diff
@@ -288,8 +288,8 @@ export class Collection<TSchema extends Document = Document> {
...
+    // specifically to an insertOne call, so we want to just support this here
```
Comment (dariakp) [inline src/collection.ts:292]: we may want to specify which "older" version this is true for

File: src/mongo_client.ts [hunk 7]
```diff
@@ -243,7 +243,7 @@ export interface MongoClientOptions extends BSONSerializeOptions, SupportedNodeC
   /** Server API version */
   serverApi?: ServerApi | ServerApiVersion;
   /**
-   * Optionally enable client side auto encryption
+   * Optionally enable in use auto encryption
```
Comment (dariakp) [inline src/mongo_client.ts:246]:
```suggestion
   * Optionally enable in-use auto encryption
```

**Review by dariakp — CHANGES_REQUESTED** [review]

**Pull request merged**